### ISSUE TITLE
Is react 16 supported?

### ISSUE THREAD

**Issue by kobyl** [bencompton/framework7-react#69]
Hello,

  I recently tried upgrading to react 16 and immediately ran into an 'object' is undefined error when importing Framework7App.

  Is react 16 supported yet?

Thanks!

Koby

**Comment by belonious** [comment]
same here.
tried `import {object} from "prop-types";` and it worked 
but then i got many warnings about unique keys.

**Comment by kobyl** [comment]
Ah, did yours render though? I didn't get past the object error.

**Comment by belonious** [comment]
yeah it renders

**Comment by kobyl** [comment]
thx! For the keys thing - did you try adding key={<someUniqueKey>} for your lists?

**Comment by jascolley** [comment]
You are correct @kobyl. You just need to make sure the key you add to each list item is unique. That should resolve the key warnings you are running into.

**Comment by kobyl** [comment]
Where did you add the import { object } from prop-types; ?

I added it to the file I was invoking Framework7App, but it didn't seem to help. Are you doing a custom build of f7-react?

Thx!

**Comment by belonious** [comment]
@kobyl 
I cloned [sample app](https://github.com/bencompton/framework7-react-app-template) and then I migrated it to TS.

As i said i had to `import {object} from "prop-types";`
for `LeftPanel.contextTypes = {
    framework7AppContext: object
};`
It works fine with these packs:
 ` {"framework7": "^1.6.5", 
    "framework7-react": "^0.9.1", 
    "react": "^16.0.10", 
    "react-dom": "^16.0.10", 
    "react-scripts-ts": "2.7.0", 
    "prop-types": "^15.5.10"}`

but I get warnings(not from my components) 

`Warning: Each child in an array or iterator should have a unique "key" prop. See https://fb.me/react-warning-keys for more information.
    in div
    in F7Navbar`

but from all F7's components.

**Comment by kobyl** [comment]
Thanks! I'm an idiot - turns out I was using an older version of framework7-react that didn't have the import in its tsx file.

All good after upgrading - thx!

**Comment by EleRam** [comment]
I upgraded to React 16 and now I have a red console! (too many warnings).


![image](https://user-images.githubusercontent.com/3352418/31608413-202cf662-b27d-11e7-8f44-59a1cc88c530.png)

**Comment by belonious** [comment]
yeap same thing

**Comment by jascolley** [comment]
The warning shown in the pic can be resolved simply by adding the unique key prop to any list items you are rendering. Just add key={someUniqueValue} to each individual list item. The framework has not been updated yet to support React 16. Hope this helps!

**Comment by EleRam** [comment]
@jascolley thanks.
I tried that and the same result happened.
